LinkedList/LinkedList_practice.py

```diff
@@ -0,0 +1,172 @@
+import pytest
+# Add typing to all methods
+# Add unit tests for all methods
+# Write time and space complexity for all methods as comment in their docstring
+
+class Node:
+    def __init__(self, data) -> None: 
+        """
+        Initializes a new node with the specified data.
+        Parameters:
+        - data: The data to store in the node.
+
+        Time complexity - O(1)
+        Space complexity - O(1)
+        """
+        self.data = data
+        self.next = None
+    
+
+class LinkedList:
+    def __init__(self) -> None:
+        """
+        Initializes a new empty linked list.
+        """
+        self.head = None
+
+    def insert(self, data) -> None:
```

Comment:
Can add typing for data as well.
```suggestion
from typing import Any (on top of file)

    def insert(self, data: Any) -> None:
```